### PR TITLE
addFilterToQuery tests

### PR DESCRIPTION
**Pull request by mxstbr**
~~**WIP, do not merge yet**~~

Adding tests for all existing `addFilterToQuery` methods, ref #1776.

- [x] boolean
- [x] code
- [x] color
- [x] date 
- [x] email
- [x] html
- [x] key
- [x] money
- [x] number
- [x] location
- [x] password
- [x] relationship
- [x] select
- [x] text
- [x] textarea
- [x] url

*(EDIT: 62 tests in one PR hell yea)*

### CODE REVIEW

File: fields/types/text/TextType.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ text.prototype.addFilterToQuery = function (filter, query) {
 	query = query || {};
 	if (filter.mode === 'exactly' && !filter.value) {
 		query[this.path] = filter.inverted ? { $nin: ['', null] } : { $in: ['', null] };
-		return;
+		return query;
```
Comment (mxstbr) [inline fields/types/text/TextType.js]: @JedWatson I'm not sure but I think this was omitted on accident, changing `query` but then not returning it seems like a bug.

I added tests for this, but if this is the intended behaviour I'll happily revoke this change and update the tests accordingly.

**Comment by mxstbr** [comment]
Just noticed the `number` field also has an `addFilterToQuery` method – will do that too before we merge!

File: fields/types/number/NumberType.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ number.prototype.addFilterToQuery = function (filter, query) {
 	query = query || {};
 	if (filter.mode === 'equals' && !filter.value) {
 		query[this.path] = filter.inverted ? { $nin: ['', 0, null] } : { $in: ['', 0, null] };
-		return;
+		return query;
```
Comment (mxstbr) [inline fields/types/number/NumberType.js]: Same question as above for the text field.

File: fields/types/number/NumberType.js [hunk 3]
```diff
@@ -59,7 +59,7 @@ number.prototype.addFilterToQuery = function (filter, query) {
 		if (!isNaN(min) && !isNaN(max)) {
 			query[this.path] = filter.inverted ? { $gte: max, $lte: min } : { $gte: min, $lte: max };
 		}
-		return;
+		return query;
```
Comment (mxstbr) [inline fields/types/number/NumberType.js]: Same question as above for the text field.

**Comment by mxstbr** [comment]
- Money field now uses number field `addFilterToQuery´ (tested this in the admin ui and it works)

**Comment by JedWatson** [comment]
For the record: **so** stoked with this PR

**Pull request merged**